test/tasks/ods-start_test.go

```diff
@@ -48,58 +51,92 @@ func TestTaskODSStart(t *testing.T) {
+					subrepoContext = subCtxt
+					err = os.RemoveAll(tempDir)
+					if err != nil {
+						t.Fatal(err)
+					}
+					err = createStartODSYML(wsDir, filepath.Base(tempDir))
+					if err != nil {
+						t.Fatal(err)
+					}
+					ctxt.ODS = tasktesting.SetupBitbucketRepo(t, ctxt.Clients.KubernetesClientSet, ctxt.Namespace, wsDir, bitbucketProjectKey)
+
+					nexusClient := nexusClientOrFatal(t, ctxt.Clients.KubernetesClientSet, ctxt.Namespace)
+					groupBase := fmt.Sprintf("/%s/%s/%s", subCtxt.Project, subCtxt.Repository, subCtxt.GitCommitSHA)
+					artifactsBaseDir := filepath.Join(projectpath.Root, "test", tasktesting.TestdataWorkspacesPath, "hello-world-app-with-artifacts", pipelinectxt.ArtifactsPath)
+					err = nexusClient.Upload(groupBase+"/xunit-reports", filepath.Join(artifactsBaseDir, "xunit-reports", "report.xml"))
+					if err != nil {
+						t.Fatal(err)
+					}
+					err = nexusClient.Upload(groupBase+"/pipeline-runs", filepath.Join(artifactsBaseDir, "pipeline-runs", "foo-zh9gt0.json"))
+					if err != nil {
+						t.Fatal(err)
+					}
 
-			// 		checkODSFileContent(t, wsDir, "component", ctxt.ODS.Component)
-			// 		checkODSFileContent(t, wsDir, "git-commit-sha", ctxt.ODS.GitCommitSHA)
-			// 		checkODSFileContent(t, wsDir, "git-full-ref", ctxt.ODS.GitFullRef)
-			// 		checkODSFileContent(t, wsDir, "git-ref", ctxt.ODS.GitRef)
-			// 		checkODSFileContent(t, wsDir, "git-url", ctxt.ODS.GitURL)
-			// 		checkODSFileContent(t, wsDir, "namespace", ctxt.Namespace)
-			// 		checkODSFileContent(t, wsDir, "pr-base", "")
-			// 		checkODSFileContent(t, wsDir, "pr-key", "")
-			// 		checkODSFileContent(t, wsDir, "project", ctxt.ODS.Project)
-			// 		checkODSFileContent(t, wsDir, "repository", ctxt.ODS.Repository)
+					ctxt.Params = map[string]string{
+						"url":               ctxt.ODS.GitURL,
+						"git-full-ref":      "refs/heads/master",
+						"project":           ctxt.ODS.Project,
+						"environment":       ctxt.ODS.Environment,
+						"version":           ctxt.ODS.Version,
+						"pipeline-run-name": "foo",
+					}
+				},
+				WantRunSuccess: true,
+				PostRunFunc: func(t *testing.T, ctxt *tasktesting.TaskRunContext) {
+					wsDir := ctxt.Workspaces["source"]
+
+					// Check .ods directory contents of main repo
+					checkODSFileContent(t, wsDir, "component", ctxt.ODS.Component)
+					checkODSFileContent(t, wsDir, "git-commit-sha", ctxt.ODS.GitCommitSHA)
+					checkODSFileContent(t, wsDir, "git-full-ref", ctxt.ODS.GitFullRef)
+					checkODSFileContent(t, wsDir, "git-ref", ctxt.ODS.GitRef)
+					checkODSFileContent(t, wsDir, "git-url", ctxt.ODS.GitURL)
+					checkODSFileContent(t, wsDir, "namespace", ctxt.Namespace)
+					checkODSFileContent(t, wsDir, "pr-base", "")
+					checkODSFileContent(t, wsDir, "pr-key", "")
+					checkODSFileContent(t, wsDir, "project", ctxt.ODS.Project)
+					checkODSFileContent(t, wsDir, "repository", ctxt.ODS.Repository)
```

Comment:
This is something we can wrap into a `checkODSDirectoryContents` func in `common_test.go` and pass the `wsDir` and `ctxt`. This way, we can reuse it for the subrepos and avoid code duplication in the test file.

Reply:
Done in fc86721.